src/utils/events/event-manager.js

```diff
@@ -0,0 +1,170 @@
+      // If `event` is a map, call `on()` for each entry.
+      for (const eventName in event) {
+        this.on(eventName, event[eventName]);
+      }
+    }
+  }
+
+  /**
+   * Deregister a previously-registered event handler.
+   * @param {string|Object} event   An event name (String) or map of event names to handlers
+   * @param {Function} [handler]    The function to be called on `event`.
+   */
+  off(event, handler) {
+    if (typeof event === 'string') {
+      // Clean up aliased gesture handler as necessary.
+      const recognizerEvent = EVENT_RECOGNIZER_MAP[event];
+      if (recognizerEvent) {
+        const eventAlias = GESTURE_EVENT_ALIASES[event];
+        if (eventAlias && this.aliasedEventHandlers[event]) {
+          this.manager.off(eventAlias, this.aliasedEventHandlers[event]);
+          delete this.aliasedEventHandlers[event];
+        }
+      }
+
+      // Deregister event handler.
+      this.manager.off(event, handler);
+    } else {
+      // If `event` is a map, call `off()` for each entry.
+      for (const eventName in event) {
+        this.off(eventName, event[eventName]);
+      }
+    }
+  }
+
+  /**
+   * Handle basic events using the 'hammer.input' Hammer.js API:
+   * Before running Recognizers, Hammer emits a 'hammer.input' event
+   * with the basic event info. This function emits all basic events
+   * aliased to the "class" of event received.
+   * See constants.BASIC_EVENT_CLASSES basic event class definitions.
+   */
+  _onBasicInput(event) {
+    const {srcEvent} = event;
+    const eventAliases = BASIC_EVENT_ALIASES[srcEvent.type];
+    if (eventAliases) {
+      // fire all events aliased to srcEvent.type
+      eventAliases.forEach(alias => {
+        const emitEvent = Object.assign({}, event, {type: alias});
+        this.manager.emit(alias, emitEvent);
+      });
+    }
+  }
+
+  /**
+   * Handle events not supported by Hammer.js,
+   * and pipe back out through same (Hammer) channel used by other events.
+   */
+  _onOtherEvent(event) {
+    const {srcEvent: {type}} = event;
+    this.manager.emit(type, event);
+  }
+
+  /**
+   * Alias one event name to another,
+   * to support events supported by Hammer.js under a different name.
+   * See constants.GESTURE_EVENT_ALIASES.
+   */
+  _aliasEventHandler(eventAlias) {
+    return event => this.manager.emit(eventAlias, event);
+  }
+
+  /**
+   * If single and double tap are both enabled,
+   * The single tap recognizer must wait for the double tap recognizer
+   * to fail before resolving. Note that enabling both incurs a slight delay
+   * on tap/click handler resolution.
+   */
+  _reconcileSingleAndDoubleTap(event) {
+    if (event === 'tap' || event === 'doubletap') {
+      const singletapRecognizer = this.manager.get('tap');
+      const doubletapRecognizer = this.manager.get('doubletap');
+      if (singletapRecognizer.options.enable && doubletapRecognizer.options.enable) {
+        singletapRecognizer.requireFailure('doubletap');
```

Comment:
My concern was that requiring failure of a disabled recognizer would cause the single tap to never fire. Not clear on a skim of hammer's source if this is the case, or if hammer is smart enough to skip the disabled recognizer in the `requireFail` check...

Though I _think_ [this](https://github.com/hammerjs/hammer.js/blob/master/src/recognizerjs/recognizer-constructor.js#L233) means that if the fail-required recognizer hasn't started running yet (which is the case when `enable: false`) it will be skipped.

Which means you're right, I can just do this via config. Let me test and circle back.